src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,53 +1,109 @@
 package core.basesyntax;
 
-/**
- * <p>Реалізувати свій ArrayList який імплементує інтерфейс List. Дотриматися основних вимог щодо
- * реалізації ArrayList (default capacity, newCapacity...)</p>
- */
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] elementData;
+    private int size;
+    private int capacity;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    public ArrayList(int capacity) {
+        this.capacity = capacity;
+        elementData = (T[]) new Object[capacity];
+    }
+
+    public void ensureCapacity(int minCapacity) {
+        if (minCapacity > capacity) {
+            capacity = (minCapacity * 3) / 2 + 1;
+            Object[] oldData = elementData;
+            elementData = (T[]) new Object[capacity];
+            System.arraycopy(oldData, 0, elementData, 0, size - 1);
+        }
+    }
+
+    private boolean checkIndex(int index) {
+        if (index >= size) {
+            throw new ArrayIndexOutOfBoundsException("Wrong index of array");
+        }
+        return true;
+    }
 
     @Override
     public void add(T value) {
-
+        size++;
+        ensureCapacity(size);
+        elementData[size - 1] = value;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        checkIndex(index);
```

Comment:
you are making the same check twice

Reply:
It is not the same check.
Method add(T value, int index) is the only method for which (index = size) is valid. For all others should be (index < size). My method checkIndex() is universal for all methods except add(T value, int index). So I added one additional check to this method.

Reply:
agree